api/v1alpha1/artifact_types.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2024 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"time"
+
+	v1 "github.com/fluxcd/source-controller/api/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// BucketKind is the string representation of a Bucket.
+	ArtifactKind = "Artifact"
+)
+
+// ArtifactSpec defines the desired state of Artifact
+type ArtifactSpec struct {
+
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	CurrentVersion string `json:"currentVersion,omitempty"`
+
+	// +kubebuilder:validation:Required
+	Versions map[string]*v1.Artifact `json:"versions,omitempty"`
+}
+
+// ArtifactStatus defines the observed state of Artifact
+type ArtifactStatus struct {
+}
```

Comment:
```suggestion
type ArtifactStatus struct {
	// +kubebuilder:validation:Required
	History []v1.Artifact `json:"history"`
}
```

This is how we track versions in Flux APIs, the first entry is the current version, all this logic is already in helm-controller.